gaia/core/task.py

```diff
@@ -121,11 +169,9 @@ def run(self, *arg, **kw):
         the input ports are all connected and raises an error if they
         aren't.
         """
-        for name in self.input_ports:
-            itask = self.get_input_task(name)
-            if itask is None:
-                raise Exception("Input port '{0}' not connected.".format(name))
-            self._input_data[name] = itask.get_output_data(name)
+        self.dirty = False
```

Comment:
We may have to move to start using timestamps but that would be in a separate PR